users/migrations/0012_allow_duplicate_account_emailaddresses.py

```diff
@@ -0,0 +1,21 @@
+from django.db import migrations
+
+class Migration(migrations.Migration):
```

Comment:
We should check if the constraint exists before trying to remove it. Would something like this work:

```
from django.db import migrations, models
from django.db.utils import OperationalError

def check_constraint_exists(apps, schema_editor):
    # Get the connection to the database
    connection = schema_editor.connection

    constraint_name = 'account_emailaddress_email_key'
    table_name = 'account_emailaddress'

    # Check if the constraint exists
    sql = f"""
    SELECT conname
    FROM pg_constraint
    WHERE conname = '{constraint_name}' AND conrelid = '{table_name}'::regclass;
    """

    with connection.cursor() as cursor:
        cursor.execute(sql)
        result = cursor.fetchone()
        return result is not None

def modify_constraints(apps, schema_editor):
    if check_constraint_exists(apps, schema_editor):
        # Remove the existing unique constraint on 'email'
        schema_editor.remove_constraint(
            model_name='emailaddress',
            name='account_emailaddress_email_key',
        )
    
    # Add the new unique constraint on ('user', 'email')
    schema_editor.add_constraint(
        model_name='emailaddress',
        constraint=models.UniqueConstraint(
            fields=['user', 'email'],
            name='account_emailaddress_email_key',
        ),
    )

class Migration(migrations.Migration):

    dependencies = [
        ('account', '0002_email_max_length'),
        ('users', '0011_change_last_name_length'),
    ]

    operations = [
        migrations.RunPython(modify_constraints),
    ]
```

Reply:
I was thinking something like this:
```
from django.db import migrations

class Migration(migrations.Migration):

    dependencies = [
        ('account', '0002_email_max_length'),
        ('users', '0011_change_last_name_length'),
    ]

    operations = [
        migrations.RunSQL(
            sql=[
                """
                DO $$ 
                DECLARE
                    r RECORD;
                BEGIN
                    FOR r IN 
                        SELECT conname 
                        FROM pg_constraint
                        JOIN pg_class ON conrelid = pg_class.oid
                        WHERE pg_class.relname = 'account_emailaddress'
                            AND conname LIKE '%email%'
                            AND contype = 'u'
                    LOOP
                        EXECUTE 'ALTER TABLE account_emailaddress DROP CONSTRAINT ' || r.conname;
                    END LOOP;
                END $$;
                """,
                "ALTER TABLE account_emailaddress ADD CONSTRAINT account_emailaddress_email_key UNIQUE (user_id, email);",
            ],
            reverse_sql=[
                "ALTER TABLE account_emailaddress DROP CONSTRAINT account_emailaddress_email_key;",
                "ALTER TABLE account_emailaddress ADD CONSTRAINT account_emailaddress_email_key UNIQUE (email);",
            ],
        ),
    ]

```